doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+
+2. Switch to the "Advanced" tab and click on "Environment Variables..."
+
+3. Double click on ``PATH`` and add ``;C:\ghcup\bin`` at the end of the already existing value
+
+Invoking GHC on Windows
+-----------------------
+
+GHC for Windows is shipped with a MinGW toolchain included. The set of included
+packages can be found in `the GHC downloads page
+<https://downloads.haskell.org/ghc/mingw>`_. Starting on version 0.7 of the
+MinGW toolchain, the included packages are based on the ``CLANG64`` environment.
+This built-in toolchain allows GHC to run on native Windows by using the bundled
+``clang`` compiler.
+
+Once GHCup installed GHC, it is possible to invoke just it from any Windows
+shell (such as PowerShell or ``cmd``) as long as the ``ghc.exe`` executable is
+in the ``PATH``. When running ``ghc --info``, it can be seen that GHC already
+sets flags for includes and library paths for the bundled toolchain:
+
+::
+
+    > ghc --info
+    [("Project name","The Glorious Glasgow Haskell Compilation System")
+    ,("C compiler command","C:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/bin/clang.exe")
+    ,("C compiler flags","--rtlib=compiler-rt -D_UCRT -Qunused-arguments -IC:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/include")
+    ,("C++ compiler command","C:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/bin/clang++.exe")
+    ,("C++ compiler flags","-IC:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/include")
+    ,("C compiler link flags","-fuse-ld=lld --rtlib=compiler-rt -D_UCRT  -LC:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/lib -LC:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/x86_64-w64-mingw32/lib")
+    ,("C compiler supports -no-pie","NO")
+    ,("Haskell CPP command","C:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/bin/clang.exe")
+    ,("Haskell CPP flags","-E -undef -traditional -Wno-invalid-pp-token -Wno-unicode -Wno-trigraphs -IC:\\ghcup\\ghc\\9.8.1\\lib\\../mingw/include")
+    ...
+
+However this is of limited use because GHC will only be able to invoke tools
+that are bundled in the installation. It won't be able to call things like
+``make`` or ``autoconf`` which are usually needed for configuring packages, and
+it will also not be able to see any third-party libraries that are installed in
+the system.
+
+.. _Using system libraries with GHC:
+
+Using system libraries with GHC
+-------------------------------
+
+Packages can be installed in an MSYS2 environment via the ``pacman`` package
+manager. The installation of those packages is done in standard directories in
+the system, namely:
+
+- ``C:\msys64\usr\{bin,include,lib}``: for basic POSIX-like utilities. We might
+  use tools that are in ``C:\msys64\usr\bin`` but we should not link against
+  libraries in ``C:\msys64\usr\lib`` as those target ``msys-2.0.dll`` as a
+  runtime dependency which is a fork of Cygwin and not a native Windows DLL.
+  These are referred to as belonging to the ``msys`` repository.
+
+  In order to use tools from ``C:\msys64\usr\bin`` we need to make sure it is
+  present in the ``PATH``.

```

Comment:
(github swallowed this comment, so here it is again)

This is not recommended, because changing PATH affects the dynamic link library search path for all other tooling as well: https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order#standard-search-order-for-unpackaged-apps

Hence, we should rely on `extra-prog-path` and invoke things via cabal or in a proper msys2 shell.

Reply:
Completely agreed with this. There are a lot of resources out there which recommend changing PATH, and I’ve never understood why, when running through the MSYS2 shell is so much easier and more robust. (See also my mention of `mintty` above.)

Reply:
Well there is a discrepancy here. What if I want to compile something on PowerShell and not on an MSYS2 shell? Isn't one of the points of GHC bundling a MINGW toolchain allowing the user to call it from PowerShell and not rely on a system's MSYS2? What if I need to call `make` from PowerShell?

I just presented what needs to happen, those directories need to be in the PATH, either by setting the env var or by telling cabal to dynamically extend the path with them when calling other tools. Setting the env var indeed will impact other things.

Anyways, this section is now gone. In the cabal documentation I now just mention how to run Cabal, which means that there only a need to modify `extra-prog-path` and nothing else.

Reply:
> What if I want to compile something on PowerShell and not on an MSYS2 shell?

Then, if really necessary, you can change the PATH temporarily. But I’ve almost never found it necessary — firing up one of the shells installed with MSYS2 is much more convenient.

Reply:
> What if I need to call make from PowerShell?

`ghcup run -m -- make`.

But how is that a common use case?

Reply:
Admittedly not that common, but I was just trying to make sure the UX was smoother by _suggesting_ the PATH modification, for a seamlessly integration in PS.

Reply:
Only after this PR https://github.com/haskell/cabal/pull/9527